lib/src/main/java/uk/gov/di/ipv/cri/kbv/api/service/ServiceFactory.java

```diff
@@ -62,4 +73,28 @@ public AuditService getAuditService() {
                 new ObjectMapper(),
                 new AuditEventFactory(getConfigurationService(), Clock.systemDefaultZone()));
     }
+
+    public KBVGateway getKbvGateway() {
+        if (this.kbvGateway == null) {
+            this.kbvGateway =
+                    getKbvGateway(
+                            new KeyStoreLoader(getConfigurationService()), getKbvClientFactory());
+        }
+        return this.kbvGateway;
+    }
+
+    KBVGateway getKbvGateway(KeyStoreLoader keyStoreLoader, KBVClientFactory kbvClientFactory) {
+
+        return new KBVGatewayFactory(keyStoreLoader, kbvClientFactory, getConfigurationService())
+                .create();
+    }
+
+    private KBVClientFactory getKbvClientFactory() {
+        var soapToken = new SoapToken("GDS DI", true, new TokenService(), configurationService);
+        var provider = new HeaderHandlerProvider(new Base64TokenCacheLoader(soapToken));
+        var headerResolver = new HeaderHandlerResolver(provider.getHeaderHandler());
```

Comment:
I like `var` but I think we said before we'd only use it in tests and not in production code